openstack/networking/v2/extensions/lbaas_v2/pools/requests.go

```diff
@@ -110,15 +116,33 @@ type CreateOpts struct {
 // Create accepts a CreateOpts struct and uses the values to create a new
 // load balancer pool.
 func Create(c *gophercloud.ServiceClient, opts CreateOpts) CreateResult {
+	var res CreateResult
+
+	// Validate inputs
```

Comment:
1) we'd rather fail early than make a user send an HTTP request to find out they did something wrong,
2) all too often, OpenStack returns unhelpful error messages,
3) I agree, logic like this shouldn't be hard-coded in the request functions. @dagnello Have a look here for an example of how this should be done in Gophercloud: https://github.com/rackspace/gophercloud/blob/master/openstack/networking/v2/networks/requests.go#L101 . All the operations in the packages should follow that design.